synchronization/src/main/kotlin/de/cyface/synchronization/SyncAdapter.kt

```diff
@@ -641,7 +640,9 @@ class SyncAdapter private constructor(
             }
             val filesSize = if (allAttachments > 0) {
                 // TODO: support multiple attachments by zipping them
-                val attachment = persistence.attachmentDao!!.loadOneByMeasurementIdAndType(measurement.id, FileType.JSON)
+                // (!) we load the CSV file here as the location_metrics.csv should always exist
+                // if not, the upload crashes with an NPE.
+                val attachment = persistence.attachmentDao!!.loadOneByMeasurementIdAndType(measurement.id, FileType.CSV)
```

Comment:
fix: we now always create a CSV file but the JSON file is optional, which was the cause for the NPE. This is a hot fix as we're planning to zip all meta attachment files anyway soon